pkg/config/squash.go

```diff
@@ -123,64 +141,113 @@ func (s *Squash) getDebuggerPodNamespace() string {
 	return s.Namespace
 }
 
-func (s *Squash) ReportOrConnectToCreatedDebuggerPod(createdPod *v1.Pod) error {
+func (s *Squash) ReportOrConnectToCreatedDebuggerPod() error {
 	if s.Machine {
-		fmt.Printf("pod.name: %v", createdPod.Name)
+		// fmt.Printf("pod.name: %v", createdPod.Name)
 	} else {
-		return s.connectUser(createdPod)
+		return s.connectUser()
 	}
 	return nil
 }
 
-func (s *Squash) connectUser(createdPod *v1.Pod) error {
+// TODO - remove this when V2 api is ready
+func (s *Squash) getIntent() squashv1.Intent {
+	return squashv1.Intent{
+		Debugger: s.Debugger,
+		Pod: &core.ResourceRef{
+			Name:      s.Pod,
+			Namespace: s.Namespace,
+		},
+		ContainerName: s.Container,
+	}
+}
+
+func (s *Squash) connectUser() error {
 	if s.Machine {
 		return nil
 	}
-	// Starting port forward in background.
-	portSpec := fmt.Sprintf("%v:%v", s.LocalPort, sqOpts.DebuggerPort)
-	cmd1 := exec.Command("kubectl", "port-forward", createdPod.ObjectMeta.Name, portSpec, "-n", s.getDebuggerPodNamespace())
-	cmd1.Stdout = os.Stdout
-	cmd1.Stderr = os.Stderr
-	cmd1.Stdin = os.Stdin
-	err := cmd1.Start()
+	debugger := local.GetParticularDebugger(s.Debugger)
+	// Refactor - eventually Intent will be created during config/user entry
+	intent := s.getIntent()
+	daClient, err := utils.GetDebugAttachmentClient(context.Background())
 	if err != nil {
-		s.showLogs(err, createdPod)
+		return err
+	}
+	da, err := intent.GetDebugAttachment(daClient)
+	if err != nil {
+		return err
+	}
+	remoteDbgPort, err := local.GetDebugPortFromCrd(da.Metadata.Name, s.Namespace)
+	if err != nil {
+		return err
+	}
+	kubectlCmd := debugger.GetRemoteConnectionCmd(
+		da.Attachment,
+		s.SquashNamespace,
+		s.Pod,
+		s.Namespace,
+		s.LocalPort,
+		remoteDbgPort,
+	)
+	// Starting port forward in background.
+	if err := kubectlCmd.Start(); err != nil {
+		// s.printError(createdPodName)
 		return err
 	}
 	// kill the kubectl port-forward process on exit to free the port
 	// this defer must be called after Start() initializes Process
-	defer cmd1.Process.Kill()
+	defer kubectlCmd.Process.Kill()
 
 	// Delaying to allow port forwarding to complete.
-	duration := time.Duration(5) * time.Second
-	time.Sleep(duration)
-
-	// TODO(mitchdraft) dlv only atm - check if dlv before doing this
-	cmd2 := exec.Command("dlv", "connect", fmt.Sprintf("127.0.0.1:%v", s.LocalPort))
-	cmd2.Stdout = os.Stdout
-	cmd2.Stderr = os.Stderr
-	cmd2.Stdin = os.Stdin
-	err = cmd2.Run()
-	if err != nil {
-		log.Warn("failed, printing logs")
-		log.Warn(err)
-		s.showLogs(err, createdPod)
+	time.Sleep(5 * time.Second)
+	if os.Getenv("DEBUG_SELF") != "" {
```

Comment:
re-enable this in a followup.

Reply:
https://github.com/solo-io/squash/issues/88